tests/client_test.py

```diff
@@ -762,6 +763,52 @@ def worker_handler(msg):
     self.assertEqual(1, err_count)
 
 
+class TLSTest(TLSServerTestCase):
```

Comment:
would be nice to add some reconnection tests with TLS context as well (can add in different PR)